web/src/stream_edit.js

```diff
@@ -286,6 +287,29 @@ export function update_muting_rendering(sub) {
     $edit_container.find(".mute-note").toggleClass("hide-mute-note", !sub.is_muted);
 }
 
+function stream_notification_reset(e) {
+    const sub = get_sub_for_target(e.target);
+    if (!sub) {
+        blueslip.error("undefined sub in stream_notification_reset()");
+        return;
+    }
+    const data = [{stream_id: sub.stream_id, property: "is_muted", value: false}];
+    for (const [setting, option] of Object.entries(
+        settings_config.generalize_stream_notification_setting,
+    )) {
+        data.push({
+            stream_id: sub.stream_id,
+            property: setting,
+            value: realm_user_settings_defaults[option],
```

Comment:
This is the wrong thing to reset it to! We want to reset it to your own personal default setting for streams, not the organization-level default for new users!

You want to access `user_settings[]` here.